extensions/blocks/story/edit.native.js

```diff
@@ -24,16 +31,99 @@ import {
 import styles from './editor.scss';
 import StoryEditingButton from './story-editing-button';
 
-const StoryEdit = ( {
-		attributes,
-		isSelected,
-		clientId,
-		editButton = true,
-} ) => {
-	const { mediaFiles } = attributes;
-	const hasContent = !! mediaFiles.length;
+class StoryEdit extends React.Component {
```

Comment:
@mzorz just checking. Why did you convert from using a function component to a class component? The function component approach eliminates `componentDidMount` , `componentWillUnmount `, `this.state = {` with hooks such as `useEffect` , `useState` etc. If it's a case where you are more familiar with the class component so we would keep it as such for now and then refactor later?

Reply:
Good question! It started as a function because all I cared about was to have a way to display it, but then in this second set of PRs I realized I needed the lifecycle of the component (specifically `componentDidMount`) to be able to call `mediaUploadSync()` and `mediaSaveSync()` bridge methods to attach the listeners when in-progress saves or uploads are happening at the beginning. It's definitely more familiar to me and also went the way of the existing Image (see class ImageEdit) or Video (see VideoEdit) blocks given I needed to reflect very similar states (like, uploading and saving).
I didn't know about function components so, definitely worth looking into that at a later refactor. For now I'd keep blocks with a similar mission written in a similar fashion so it's easier to navigate code and compare one another, if that makes sense. That said, it'd definitely make sense to analyze the common behaviors and structures in the Video, Image, Story, Gallery blocks (and any other block utilizing or representing some kind of media arrangement)  and refactor when it makes sense.
I'd rather not break other blocks for now though 😅  lol

Reply:
Okay understood. Thanks for expounding in such detail. Yes, the familiarity and ease of implementing the logic similarly to other media related blocks that share similar functionality makes sense at this time 😎

Reply:
Just to follow up on this: PR with function rewrite here https://github.com/Automattic/jetpack/pull/17549